src/writer/core.py

```diff
@@ -659,6 +678,48 @@ def _set_state_item(self, key: str, value: Any):
                 self._state_proxy[key] = value
 
 
+    def subscribe_mutation(self, path: Union[str, List[str]], handler: Callable[['State'], None], initial_triggered: bool = False) -> None:
```

Comment:
If you want to support array, the `path` should be able to contain number since string can't be index

```python
>>> obj = {"a": [1,2,3]}
>>> obj["a"]["0"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: list indices must be integers or slices, not str
```